scio-test/src/test/scala/com/spotify/scio/testing/JobTestTest.scala

```diff
@@ -733,12 +753,61 @@ class JobTestTest extends PipelineSpec {
     val msg = "requirement failed: There already exists test output for TextIO(output), " +
       "currently registered outputs: [TextIO(output)]"
     the [IllegalArgumentException] thrownBy {
-    JobTest[JobWitDuplicateOutput.type]
-      .args("--output=output")
-      .output[String](TextIO("output"))(_ should containSingleValue ("does not matter"))
-      .run()
+      JobTest[JobWitDuplicateOutput.type]
+        .args("--output=output")
+        .output[String](TextIO("output"))(_ should containSingleValue ("does not matter"))
+        .run()
     } should have message msg
   }
 
+  // =======================================================================
+  // Test metrics
+  // =======================================================================
+
+  it should "pass correct metrics test" in {
+    JobTest[MetricsJob.type]
+      .counter(ScioMetrics.counter("counter"))(_.committed shouldBe Some(10))
```

Comment:
`committed` gets repeated over, is there a reason to expose `MetricValue` instead of directly the value from `committed` or maybe even value within the `Option` (if possible)?

Reply:
Not sure, I guess it's a matter of how much we expose in `JobTest`.
I'm not exposing the `Map[MetricName, V]` or the entire `ScioResult` so it's kinda somewhere in between.

Question is how much is a good balance bewteen flexibility and API elegance?

Reply:
I guess for the case of `JobTest` it's almost always a batch job and the job must've finished when evaluating counters, so it's safe to just assert on committed result?